### PR TITLE
Fix #4940: scroll block applied incorrectly in some cases

### LINKED ISSUE #4940: Dialog: blockScroll + visible + fullscreen mode gives unexpected behavior if dialog is unmounted while visible
### Describe the bug

Quite similar issue as this: #4938 , but with more use cases that leads to bugs with scroll.
Note: despite #4938 is closed - fixes for this ticket are not enough to fix all bugs that are listed here

In general - `blockScroll' and fullscreen mode affects scroll on main page incorrectly if modal get unmounted while visible

To be precise:

1. If Dialog with `blockScroll=true` and `visible=true` get unmounted - scroll on main page disappears forever. (fixed in #4938)
2. If Dialog with `maximizable=true` and `visible=true` get unmounted while in fullscreen - scroll on main page disappears forever.
3. If several dialogs with `blockScroll=true` are displayed on the screen, after you close only one of it - scroll on main page becomes available.
4. If there is one dialog with `blockScroll=true` and another one with `maximizable=true` and `blockScroll=false`, and both are visible on the screen, after second one goes to fullscreen and returns back to normal size -  scroll on main page becomes available

See sandbox below to find live examples of such bugs

### Reproducer

https://codesandbox.io/s/primereact-test-forked-l29ctx

### PrimeReact version

9.6.2

### React version

18.x

### Language

ALL

### Build / Runtime

Create React App (CRA)

### Browser(s)

_No response_

### Steps to reproduce the behavior

See reproduction demo (as there are many steps to reproduce)

### Expected behavior

For examples provided in "Description" section:
1. scrolling should be available after dialog closed
2. scrolling should be available after fullscreen dialog closed
3. scrolling should be blocked if there is at least one modal with `blockScroll=true` is visible on the screen
4. scrolling should be blocked if there is at least one modal with `blockScroll=true` is visible on the screen

### PR DESCRIPTION
**Pull request by avasuro**
Fix #4940

### CODE REVIEW

**Review by melloware — APPROVED** [review]
This looks great.  Let me test it out.

File: components/lib/dialog/Dialog.js [hunk 1]
```diff
@@ -329,27 +326,63 @@ export const Dialog = React.forwardRef((inProps, ref) => {
...
+
+            if (currentDialogIndexInRegistry === -1) {
+                document.primeDialogParams = [...document.primeDialogParams, newParam]
+            }
+            else {
+                // Is it OK to mutate this array?
+                // If mutation is not desired - try to use modern
+                // `.toSpliced` method
+                // (see here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced)
+                document.primeDialogParams[currentDialogIndexInRegistry] = newParam
```
Comment (melloware) [inline components/lib/dialog/Dialog.js:374]: We are only supporting modern browsers so I think its OK for you to use toSpliced if its the correct thing to do?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
On more thing while you are in there my IDE is showing the method `getPositionClass` as no longer used and I just removed it and verified all the position samples are working correctly. So its something that got moved with unstyled mode but this code was not deleted. Do you mind deleting it as part of this PR please?

**Comment by avasuro** [comment]
Fixed

**Comment by melloware** [comment]
@avasuro this is a fantastic PR.  All tested and looks good on my end letting the CI Pipeline run then I will merge!

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
ahh looks like you need to run `npm run format` to format the code.

**Comment by melloware** [comment]
The failures are unrelated to this change.  Merging!

**Comment by ulasturann** [comment]
Hi @avasuro,

Thank you for your contribution through your PR. We've realized that the 'blockScroll' issue isn't related to your PR and have already addressed it. Could you please reopen your PR so that we can incorporate your valuable contribution into v10?

**Comment by melloware** [comment]
@avasuro @ulasturann I submitted it in a new PR.

**Pull request merged**